app/workers/application_worker.rb

```diff
@@ -0,0 +1,7 @@
+class ApplicationWorker
+  include Sidekiq::Worker
+
+  def run_with_advisory_lock(model, id, &block)
```

Comment:
I wonder if you need two arguments of this, it feels that it works as a good abstraction where we're running a particular id of an item but feels a bit cumbersome for things like `DigestRun, "#{date}-#{Frequency::DAILY}"` where the id argument doesn't refer to the common understanding of an id of that model.

I wonder if you could get away with just having an optional id field and have the lock automatically use the class name.

for example:

```
def run_with_advisory_lock(id = nil, &block)
ApplicationRecord.with_advisory_lock("#{self.class}-#{id}", timeout_seconds: 0) { block.call }
end
```

then you'd not need to provide any arguments for most cases and be able to just pass in a date for daily/weekly digest initiators.